tests/drivers/gpio/gpio_basic_api/src/test_gpio_port.c

```diff
@@ -672,6 +683,7 @@ void test_gpio_port(void)
 		      "bits_logical failed");
 	zassert_equal(check_pulls(), TC_PASS,
 		      "check_pulls failed");
-	zassert_equal(pin_get_config(), TC_PASS,
-		      "pin_get_config failed");
+	if (IS_ENABLED(CONFIG_GPIO_GET_CONFIG))
+		zassert_equal(pin_get_config(), TC_PASS,
```

Comment:
In case of CONFIG_GPIO_GET_CONFIG=n, the pin_get_config might definitively be not supported in the driver, then giving compilation warning --> function is undeclared 
This is not exactly the same as 
```
#ifdef CONFIG_GPIO_GET_CONFIG
 	zassert_equal(pin_get_config(), TC_PASS,
 		      "pin_get_config failed");
#endif /* CONFIG_GPIO_GET_CONFIG */
```

Reply:
Yes, we'd better to roll back

Reply:
Should be `IS_ENABLED(..) && api->get_config != NULL`

Reply:
> pin_get_config
Updated. It seems the CONFIG_GPIO_GET_CONFIG for the function pin_get_config should be remove as well